binrw/src/helpers.rs

```diff
@@ -63,11 +63,31 @@ where
     CondFn: Fn(&T) -> bool,
     Arg: Clone,
     Ret: core::iter::FromIterator<T>,
+{
+    let read = |reader: &mut Reader, ro: &ReadOptions, args: Arg| {
+        let mut value = T::read_options(reader, ro, args.clone())?;
+        value.after_parse(reader, ro, args)?;
+        Ok(value)
+    };
+    until_with(cond, read)
+}
+
+/// Do the same as [until](binrw::helpers::until) with a custom parsing function for the inner type.
```

Comment:
An example for this + the other 2 without examples is the only request I have outside of your stated docs/tests